vignettes/update-app.Rmd

```diff
@@ -0,0 +1,355 @@
+- `f7Swipeout`: update the [swipeout configuration](https://framework7.io/docs/swipeout#swipeout-app-parameters)
+- `f7Toolbar`: update the [toolbar configuration](https://framework7.io/docs/toolbar-tabbar#toolbar-app-parameters)
+
+<br>
+
+Basically, whenever a component in the [Framework7 documentation](https://framework7.io/docs/) contains a section on "global app parameters", you can update these parameters using `updateF7App`.
+
+## Update individual components at runtime
+
+If you're not looking to update the entire app or a particular group of components, but rather a specific component, you can use `updateF7Entity`. This function allows you to update any Framework7 instance from the server:
+
+```r
+updateF7Entity(
+  id = "idtoupdate",
+    options = list(
+      # your configuration here
+    )
+)
+```
+
+For example, we could update (or reset) the content of an `f7ActionSheet`:
+
+```{r, eval=FALSE}
+library(shiny)
+library(shinyMobile)
+
+shinyApp(
+  ui = f7Page(
+    title = "Update Entity",
+    f7SingleLayout(
+      navbar = f7Navbar(title = "Update action sheet instance"),
+      f7Segment(
+        f7Button(
+          inputId = "goButton",
+          "Go!",
+          fill = FALSE,
+          outline = TRUE
+        ),
+        f7Button(
+          inputId = "update",
+          "Update config",
+          fill = FALSE,
+          outline = TRUE
+        ),
+        f7Button(
+          inputId = "reset",
+          "Reset",
+          fill = FALSE,
+          outline = TRUE
+        )
+      )
+    )
+  ),
+  server = function(input, output, session) {
+    observeEvent(input$goButton, {
+      f7ActionSheet(
+        grid = TRUE,
+        id = "action1",
+        buttons = list(
+          list(
+            text = "Notification",
+            icon = f7Icon("info"),
+            color = NULL
+          ),
+          list(
+            text = "Dialog",
+            icon = f7Icon("lightbulb_fill"),
+            color = NULL
+          )
+        )
+      )
+    })
+
+    observeEvent(input$update, {
+      updateF7Entity(
+        id = "action1",
+        options = list(
+          buttons = list(
+            list(
+              text = "Notification",
+              icon = f7Icon("info"),
+              color = NULL
+            )
+          )
+        )
+      )
+    })
+
+    observeEvent(input$reset, {
+      updateF7Entity(
+        id = "action1",
+        options = list(
+          buttons = list(
+            list(
+              text = "Notification",
+              icon = f7Icon("info"),
+              color = NULL
+            ),
+            list(
+              text = "Dialog",
+              icon = f7Icon("lightbulb_fill"),
+              color = NULL
+            )
+          )
+        )
+      )
+    })
+  }
+)
+```
+
+In the above example, we update the content of `buttons` in the `f7ActionSheet` with the id `action1`. This way, you can update the component at runtime.
+
+<br>
+
+TODO: other examples
```

Comment:
@DivadNojnarg, not really a good use case yet besides action sheet, but hopefully this will change once we start moving more input widgets to the store. Then we can update this section

Reply:
I made some changes to make it possible to update gauges and other elements with this function:

```r
shinyApp(
  ui = f7Page(
    title = "Gauges",
    f7SingleLayout(
      navbar = f7Navbar(title = "f7Gauge"),
      f7Block(
        f7Gauge(
          id = "mygauge",
          type = "semicircle",
          value = 50,
          borderColor = "#2196f3",
          borderWidth = 10,
          valueFontSize = 41,
          valueTextColor = "#2196f3",
          labelText = "amount of something"
        )
      ),
      f7Block(f7Button("update", "Update Gauge"))
    )
  ),
  server = function(input, output, session) {
    observeEvent(input$update, {
      new_val <- 75
      updateF7Entity(
        id = "mygauge",
        options = list(
          # Must be between 0 and 1
          value = new_val / 100, 
          valueText = paste0(new_val, "%"),
          labelText = "New label!"
        )
      )
    })
  }
)
```

I unfortunately can't push as I have troubles with other elements.

Reply:
There seem to be a bug in Framework7, since extending a popup config and creating a new popup on this new config, creates a destroyed instance.

Reply:
Continuing on what works:

- searchbar:
```r
cities <- names(precip)

shinyApp(
  ui = f7Page(
    title = "Expandable searchbar",
    f7SingleLayout(
      navbar = f7Navbar(
        title = "f7Searchbar with trigger",
        subNavbar = f7SubNavbar(
          f7Searchbar(id = "search1", expandable = TRUE)
        )
      ),
      f7Block(
        f7Button("update", "Update")
      ),
      f7Block(
        f7SearchbarTrigger(targetId = "search1")
      ) %>% f7HideOnSearch(),
      f7List(
        lapply(seq_along(cities), function(i) {
          f7ListItem(cities[i])
        })
      ) %>% f7Found(),
      f7Block(
        p("Nothing found")
      ) %>% f7NotFound()
    )
  ),
  server = function(input, output) {
    observeEvent(input$update, {
      updateF7Entity(
        id = "search1",
        options = list(
          disableButton = FALSE
        )
      )
    })
  }
)
```

- swiper: 
```r
shiny::shinyApp(
  ui = f7Page(
    options = list(
      dark = FALSE
    ),
    title = "Swiper",
    f7SingleLayout(
      navbar = f7Navbar(title = "f7Swiper"),
      f7Swiper(
        id = "swiper",
        lapply(1:20, \(c) {
          f7Slide(
            f7Card(
              title = sprintf("Slide %s", c)
            )
          )
        })
      )
    )
  ),
  server = function(input, output) {
    observe({
      updateF7Entity(
        "swiper",
        options = list(
          speed = 100,
          slidesPerView = 2,
          spaceBetween = 10,
          autoplay = TRUE,
          scrollbar = list(
            enabled = FALSE
          ),
          navigation = list(
            enabled = FALSE
          ),
          pagination = list(
            type = "progressbar"
          ),
          grid = list(
            fill = "columns",
            rows = 4
          ),
          thumbs = TRUE
        )
      )
    })
  }
)
```

Reply:
Fixed the issue with popup:

```r
shinyApp(
  ui = f7Page(
    title = "Popup",
    f7SingleLayout(
      navbar = f7Navbar(
        title = "f7Popup"
      ),
      f7Segment(
        f7Button("toggle", "Toggle Popup"),
        f7Button("update", "Update Popup")
      )
    )
  ),
  server = function(input, output, session) {
    output$res <- renderPrint(input$text)

    observeEvent(input$toggle, {
      f7Popup(
        id = "popup",
        title = "My first popup",
        f7Text(
          "text", "Popup content",
          "This is my first popup ever, I swear!"
        ),
        verbatimTextOutput("res")
      )
    })

    observeEvent(input$update, {
      updateF7Entity(
        id = "popup",
        options = list(
          swipeToClose = TRUE,
          animate = FALSE,
          closeOnEscape = TRUE,
          # Content must contain the popup
          # layout!!!
          content = '<div class="popup">
            <div class="view">
              <div class="page">
                <div class="navbar">
                  <div class="navbar-bg"></div>
                  <div class="navbar-inner">
                    <div class="title">Popup</div>
                    <div class="right">
                      <!-- Link to close popup -->
                      <a class="link popup-close">Close</a>
                    </div>
                  </div>
                </div>
                <div class="page-content">
                  <div class="block">New content ...</div>
                </div>
              </div>
            </div>
          </div>'
        )
      )
    })
  }
)
```